stdlib/ballerina-database/src/main/java/org/ballerinalang/database/sql/actions/AbstractSQLAction.java

```diff
@@ -277,11 +277,12 @@ protected void executeBatchUpdate(Context context, SQLDatasource datasource, Str
             updatedCount = stmt.executeBatch();
             conn.commit();
         } catch (BatchUpdateException e) {
+            conn.rollback();
```

Comment:
AFAIK once batch update fails, whether all operations in batch update are rolled back or not depends on the driver. Why should we override the driver behavior here? It would make behavior of any ballerina data client consistent but wouldn't that confuse users who have been using the drivers and know how they behave?

Reply:
IMO, Ballerina data clients are abstraction over any existing driver and data client operations/outputs  should not behave differently depending on the underlying driver. That increases the complexity and users need to check the underlying documentation to get understand exact ballerina behavior.